### PR TITLE
Refactored tests for fracture mesh generation

### PR DESCRIPTION
**Pull request by keileg**
## Proposed changes
* Moved various tests for fracture mesh generation to new locations.
* Expanded tests for structured mesh generation in fractured domains
* Maintenance and improvements of tests

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Testing (contribution related to testing of existing or new functionality).
- [x] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

**Review by IvarStefansson — CHANGES_REQUESTED** [review]
Nice work. I made some improvements, please review carefully. Also left some comments, please ask if you need me to elaborate/want to discuss.

File: tests/fracs/test_fracture_network_3d.py [hunk 1]
```diff
@@ -170,3 +179,1163 @@ def test_bounding_box(fracs_expected):
...
+    for fi, f in enumerate(fractures):
+        for g in mdg.subdomains(dim=2):
+            if g.frac_num == fi:
+                assert compare_bounding_boxes(
+                    pp.domain.bounding_box_of_point_cloud(f.pts),
+                    pp.domain.bounding_box_of_point_cloud(g.nodes),
+                )
+
+    # The bounding boxes of the constructed 1d grids will be compared to the
+    # expected values. To construct the grid bounding boxes,
```
[the remark this answers is not in the document]
Comment (keileg) [reply tests/fracs/test_fracture_network_3d.py:335]: Improved documentation.

File: tests/fracs/test_fracture_network_3d.py [hunk 2]
```diff
@@ -170,3 +179,1163 @@ def test_bounding_box(fracs_expected):
...
+        domain = pp.Domain(bbox)
+        mesh_args = {"mesh_size_frac": 1, "mesh_size_bound": 1, "mesh_size_min": 0.2}
+
+        return mesh_args, domain
+
+    def test_one_fracture_touching_one_boundary(self):
+        """A single fracture completely immersed in a boundary grid."""
+        f_1 = pp.PlaneFracture(np.array([[-1, 2, 2, -1], [0, 0, 0, 0], [-1, -1, 1, 1]]))
+        domain = _standard_domain()
+        mdg = _create_mdg([f_1], domain)
```
[the remark this answers is not in the document]
Comment (keileg) [reply tests/fracs/test_fracture_network_3d.py:1040]: Checks will be introduced.

File: tests/fracs/test_fracture_network_3d.py [hunk 3]
```diff
@@ -170,3 +179,1163 @@ def test_bounding_box(fracs_expected):
...
+    def test_one_fracture_touches_boundary_and_corner(self):
+        f_1 = pp.PlaneFracture(
+            np.array([[-2, 2, 2, -2], [-2, 1, 1, -2], [-2, -2, 2, 2]])
+        )
+        domain = _standard_domain()
+        mdg = _create_mdg([f_1], domain)
+
+
+class TestDFMMeshGenerationWithConstraints:
+    """Tests similar to some those of the mother class, but with some fractures as
```
[the remark this answers is not in the document]
Comment (keileg) [reply tests/fracs/test_fracture_network_3d.py:1087]: Updated.

**Comment by keileg** [comment]
Thanks for the improvements, @IvarStefansson, some of them slipped off my radar. I believe the changes just committed should take care of your requests.

**Review by IvarStefansson — APPROVED** [review]
Approving, provided you reviewed my changes.

**Comment by keileg** [comment]
I did (eventually); they all made sense. Thanks.

**Pull request merged**